v2/internal/testcommon/samples_tester.go

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package testcommon
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/emirpasic/gods/maps/linkedhashmap"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+const RefsPackage = "refs"
+const samplesPath = "../../config/samples"
+
+// exclusions slice contains resources to exclude from test as these resources run into
+// "runtime.notRegisteredErr" error from the Scheme.
+var exclusions = [...]string{
+        "batchaccountpool", 
+        "snapshot",
+	"sqldatabasecontainerthroughputsetting", 
+	"sqldatabasethroughputsetting",
+	"storageaccountsqueueservice",
+	"webtest", 
+	// excluding dbformysql/user as is not an ARM resource
+	"user",
+}
+
+type SamplesTester struct {
+	tc            *KubePerTestContext
+	decoder       runtime.Decoder
+	UseRandomName bool
+}
+
+type SampleObject struct {
+	SamplesMap   map[string]*linkedhashmap.Map
+	childObjects map[string]genruntime.ARMMetaObject
+}
+
+func NewSampleObject() SampleObject {
+	return SampleObject{
+		SamplesMap:   make(map[string]*linkedhashmap.Map),
+		childObjects: make(map[string]genruntime.ARMMetaObject),
+	}
+}
+
+func NewSamplesTester(tc *KubePerTestContext, useRandomName bool) *SamplesTester {
+	return &SamplesTester{
+		tc:            tc,
```

Comment:
I think it might be better if you wrote this "like a library". What I mean is that, you didn't use `tc` to do assertions (`tc.Expect`) in the methods. Just return error and bubble up errors to the caller and then do the `tc.Expect(err).ToNot(HaveOccurred())` at the top level in the test helper (`runGroupTest`).

This has a number of advantages:
1. You can just peel off the `NoSpaceNamer` and `Namespace` from `tc` rather than keeping the whole `tc` on `SamplesTester`.
2. It becomes clearer which functions can fail in this file (the ones that return error).
3. It makes this code more unit testable (not that you're necessarily going to write unit tests for your test helper, but it's good practice to be in IMO).

Basically, don't _assume_ this library is going to be used in a test context, with Gomega assertions. Ours _is_ but it's easier to just leave that up to the caller of this class.